sdk/core/core-rest-pipeline/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -219,3 +318,64 @@ export function bearerTokenAuthenticationPolicy(
     },
   };
 }
+
+/**
+ *
+ * Interface to represent a parsed challenge.
+ *
+ * @internal
+ */
+interface AuthChallenge {
+  scheme: string;
+  params: Record<string, string>;
+}
+
+/**
+ * Converts: `Bearer a="b", c="d", Pop e="f", g="h"`.
+ * Into: `[ { scheme: 'Bearer', params: { a: 'b', c: 'd' } }, { scheme: 'Pop', params: { e: 'f', g: 'h' } } ]`.
+ *
+ * @internal
+ */
+export function parseChallenges(challenges: string): AuthChallenge[] {
+  // Challenge regex seperates the string to individual challenges with different schemes in the format `Scheme a="b", c=d`
+  // The challenge regex captures parameteres with either quotes values or unquoted values
+  const challengeRegex = /(\w+)\s+((?:\w+=(?:"[^"]*"|[^,]*),?\s*)+)/g;
+  // Parameter regex captures the claims group removed from the scheme in the format `a="b"` and `c="d"`
+  // CAE challenge always have quoted parameters. For more reference, https://learn.microsoft.com/entra/identity-platform/claims-challenge
+  const paramRegex = /(\w+)="([^"]*)"/g;
+
+  const parsedChallenges: AuthChallenge[] = [];
+  let match;
+
+  // Iterate over each challenge match
+  while ((match = challengeRegex.exec(challenges)) !== null) {
+    const scheme = match[1];
+    const paramsString = match[2];
+    const params: Record<string, string> = {};
+    let paramMatch;
+
+    // Iterate over each parameter match
+    while ((paramMatch = paramRegex.exec(paramsString)) !== null) {
+      params[paramMatch[1]] = paramMatch[2];
+    }
+
+    parsedChallenges.push({ scheme, params });
+  }
+  return parsedChallenges;
+}
+
+/**
+ * Parse a pipeline response and look for a CAE challenge with "Bearer" scheme
+ * Return the value in the header without parsing the challenge
+ * @internal
+ */
+function getCaeChallengeClaims(challenges: string | undefined): string | undefined {
+  if (!challenges) {
+    return;
+  }
+  // Find all challenges present in the header
+  const parsedChallenges = parseChallenges(challenges);
+  return parsedChallenges.find(
+    (x) => x.scheme === "Bearer" && x.params.claims && x.params.error === "insufficient_claims",
```

Comment:
@minhanh-phan From the [doc](https://learn.microsoft.com/en-us/entra/identity-platform/claims-challenge?tabs=dotnet), 
I notice the `authorization_uri` is required but we didn't check that condition, is there any difference in Azure?